src/parse.ts

```diff
@@ -141,7 +148,11 @@ function findTypeByName(types: MessageDefinition[], name: string): MessageDefini
     }
     // return if the search is in the type name
     // or matches exactly if a fully-qualified name match is passed to us
-    const nameEnd = name.includes("/") ? name : `/${name}`;
+    const nameEnd = name.includes("/")
+      ? name
+      : typeNamespace
+      ? `${typeNamespace}/${name}`
+      : `/${name}`;
     return typeName.endsWith(nameEnd);
```

Comment:
This logic seems weird. Based on the comment, I would expect something more like `if (name.includes("/")) { return typeName === name } ...`

Reply:
You are right, rewrote it to make it clearer and match exactly if the type is a FQN (or can be made to a FQN when the namespace is given)